src/autocomplete.js

```diff
@@ -341,15 +345,30 @@ class Autocomplete {
 
             if (finished) {
                 // No results
-                if (!filtered.length)
+                if (!filtered.length && (!this.emptyMessage || this.autoShown))
                     return this.detach();
 
+                if (!filtered.length) {
+                    var emptyMessage = !this.autoShown && this.emptyMessage;
+                    if ( typeof emptyMessage == "function")
+                          emptyMessage = this.emptyMessage(prefix);
+                    if (emptyMessage) {
+                        var completionsForEmpty = [{
+                            caption: this.emptyMessage(prefix),
```

Comment:
Could you add a test, Showing empty and pressing enter, to add coverage for these new lines?